src/core/layertree/qgslayertreemodellegendnode.cpp

```diff
@@ -1250,21 +1250,25 @@ QSizeF QgsWmsLegendNode::drawSymbol( const QgsLegendSettings &settings, ItemCont
 
   const QImage image = getLegendGraphic();
 
-  QSize targetSize = image.size();
-  if ( settings.wmsLegendSize().width() < image.width() )
+  // Assume legend images have 72 dpi resolution
+  double px2mm = 25.4 / 72;
```

Comment:
Does using the QImage dotsPerMeterX property work here instead of the hardcoded DPI?

Reply:
Yep works!